dcm2niix/arrange.py

```diff
@@ -0,0 +1,260 @@
+        log.error(
+            (
+                "Incorrect gear input. Input archive packaging is not supported. "
+                f"Multiple subdirectories detected in archive: {subdirs}. Exiting."
+            )
+        )
+        os.sys.exit(1)
+
+    # 3. If PAR file in the archive, then adjust par/rec filenames
+    if [file for file in filelist if file.lower().endswith(".par")]:
+        adjust_parrec_filenames(dcm2niix_input_dir, dirname)
+
+    return dcm2niix_input_dir
+
+
+def setup_dcm2niix_input_dir(infile, work_dir):
+    """Create dcm2niix input directory using the filename from the input filepath."""
+    if os.path.isfile(infile):
+        filename = os.path.split(infile)[-1]
+    else:
+        # Passing subdirectory name (e.g., dicom_files)
+        filename = infile
+
+    dirname = clean_filename(filename)
+    dcm2niix_input_dir = "/".join([work_dir, dirname])
+    os.mkdir(dcm2niix_input_dir)
+
+    return dcm2niix_input_dir, dirname
+
+
+def clean_filename(filename):
+    """Clean filename for the alphanumeric filename without extensions."""
+    # Drop file extension
+    # Do not use os.path.splitext because filename not required to have an extension
+    for extension in [".zip", ".tgz", ".par", ".PAR"]:
+        if filename.endswith(extension):
+            filename = filename.replace(extension, "")
+
+    # Drop imaging extensions
+    # Do not use os.path.splitext because filename may include periods
+    for extension in [".dicom", ".dcm", ".parrec"]:
+        if filename.endswith(extension):
+            filename = filename.replace(extension, "")
+
+    # Replace non-alphanumerics with underscore and avoiding repeat underscores
+    filename = re.sub("[^0-9a-zA-Z]+", "_", filename)
+
+    # Do not end on an underscore
+    filename = filename.rstrip("_")
+
+    return filename
+
+
+def strip_prefix_ziparchive(zipObj, prefix):
+    """Modify a filepath in a zip archive without the prefix."""
+    for zipinfo in zipObj.infolist():
+
+        if (
+            zipinfo.filename.startswith(prefix)
+            and os.path.splitext(zipinfo.filename)[0] != prefix
+        ):
+
+            zipinfo.filename = zipinfo.filename[len(prefix) :]
+
+        # Only return files
+        if not zipinfo.is_dir():
+
+            yield zipinfo
+
+
+def strip_prefix_tararchive(tarObj, prefix):
+    """Modify a filepath in a tar archive without the prefix."""
+    for tarinfo in tarObj.getmembers():
+
+        if (
+            tarinfo.path.startswith(prefix)
+            and os.path.splitext(tarinfo.path)[0] != prefix
+        ):
+
+            tarinfo.path = tarinfo.path[len(prefix) + 1 :]
+
+        # Only return files
+        if tarinfo.isreg():
+
+            yield tarinfo
+
+
+def adjust_parrec_filenames(dcm2niix_input_dir, filename):
+    """Rename par/rec files with the specified filename and lowercase extension."""
+    files = glob.glob(dcm2niix_input_dir + "/**", recursive=True)
```

Comment:
would leverage pathlib Path and its glob method here are in all other instances with `/*` or `/**` to avoid hardcoding path seps

Reply:
I'm not sure what the issue is. What is the pythonic issue with this statement? What is your recommended replacement?

Reply:
@kalebdfischer I'd like to understand more about the impact of the method you present. Could you give me more details?

Reply:
It's generally better to avoid the assumption that your operating system is Posix, as it could be Windows. This won't be the case in the gear's docker environment, but would be if a user is running Windows. pathlib's Path translates path separators in the glob pattern. Example assuming you `from pathlib import Path` import statement:
```
input_path = Path(dcm2niix_input_dir)
files = [path for path in input_path.glob("**/*")]
```